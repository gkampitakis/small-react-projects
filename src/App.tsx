import React from 'react';
import Birthday from './projects/birthday/birthday';
import Tours from './projects/tours/tours';

function App () {
  return (
    <div className="App">
      <Tours />
    </div>
  );
}

export default App;
