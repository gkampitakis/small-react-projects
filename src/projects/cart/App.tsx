import React, { ReactElement } from 'react';
import { useGlobalContext } from './context';
import Navbar from './components/Navbar';
import Cart from './components/Cart';

export default function App (): ReactElement {
  const { loading } = useGlobalContext();

  return (<main className="cart_project">
    {loading &&
      <div className="loading">
        <h1>Loading ...</h1>
      </div>
    }
    {!loading && <>
      <Navbar />
      <Cart />
    </>}
  </main>);
}