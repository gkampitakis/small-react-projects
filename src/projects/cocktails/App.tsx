import React, { ReactElement, lazy, Suspense } from 'react';
import { Route, Switch, } from 'react-router-dom';
import { useBodyStyles } from '../../hooks';
import { AppProvider } from './context';
import Loading from '../../components/Loading';
import Home from './pages/Home';
import Navbar from './components/Navbar';
import './index.scss';


export default function App (): ReactElement {
  useBodyStyles({
    background: '#f1f5f8'
  });

  return (
    <AppProvider>
      <main className="cocktails_project">
        <Navbar />
        <Switch>
          <Route exact component={Home} path="/cocktails" />
          <Route exact component={() => LazyDetails('About')} path="/cocktails/about" />
          <Route exact component={() => LazyDetails('Details')} path="/cocktails/details/:id" />
        </Switch>
      </main>
    </AppProvider>
  );
}

const LazyDetails = (page: string) => {
  const Component = lazy(() => import(`./pages/${page}`));

  return (
    <Suspense fallback={<Loading />}><Component /></Suspense >
  );
}
