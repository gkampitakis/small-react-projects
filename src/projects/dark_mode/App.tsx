import React, { ReactElement, useEffect } from 'react';
import { useBodyStyles, useDocTitle, useLocalStorage } from '../../hooks';
import Article from './Article';
import data from './data';
import './index.scss';

export default function App(): ReactElement {
  useDocTitle('Dark Mode');
  useBodyStyles({
    color: 'var(--clr-font)',
    background: 'var(--clr-bcg)'
  });
  const [theme, setTheme] = useLocalStorage<string>('dark-mode', 'dark_theme');

  useEffect(() => {
    document.documentElement.className = theme;
  }, [theme]);

  function toggleTheme() {
    const themes: Record<string, string> = {
      dark_theme: 'light_theme',
      light_theme: 'dark_theme'
    };

    setTheme(themes[theme]);
  }

  return (
    <main className="dark_mode_project">
      <header>
        <div>
          <h1>overreacted</h1>
          <button onClick={toggleTheme}>toggle</button>
        </div>
      </header>
      <section className="articles">
        {data.map(({ id, ...rest }) => (
          <Article key={id} {...rest} />
        ))}
      </section>
    </main>
  );
}
