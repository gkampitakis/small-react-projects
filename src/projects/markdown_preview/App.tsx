import React, { ReactElement, useState } from 'react';
import ReactMarkdown from 'react-markdown';
import { useBodyStyles } from '../../hooks';
import 'github-markdown-css';
import './index.scss'

export default function App (): ReactElement {
  useBodyStyles({
    background: 'hsl(210, 36%, 96%)'
  });
  const [preview, setPreview] = useState('# markdown preview');

  return (
    <main className="markdown_preview">
      <section>
        <textarea
          value={preview}
          onChange={(e) => setPreview(e.target.value)}
        />
        <article>
          <ReactMarkdown className="markdown-body">{preview}</ReactMarkdown>
        </article>
      </section>
    </main>
  );
}
