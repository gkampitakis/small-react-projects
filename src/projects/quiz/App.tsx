import React, { ReactElement } from 'react';
import { AppProvider } from './context';
import Quiz from './Components/Quiz';
import './index.scss';
import { useBodyStyles, useDocTitle } from '../../hooks';

export default function App(): ReactElement {
  useBodyStyles({
    background: 'hsl(210, 36%, 96%)'
  });
  useDocTitle('Quiz');

  return (
    <AppProvider>
      <Quiz />
    </AppProvider>
  );
}
