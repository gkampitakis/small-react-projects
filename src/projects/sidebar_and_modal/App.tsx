import React from 'react';
import Home from './components/Home';
import Modal from './components/Modal';
import Sidebar from './components/Sidebar';
import { AppProvider } from './context';
import './index.css';

export default function App () {
  return (
    <AppProvider>
      <Home />
      <Modal />
      <Sidebar />
    </AppProvider>
  )
}
