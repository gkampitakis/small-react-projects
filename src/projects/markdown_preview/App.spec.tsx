import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, screen, render } from '@testing-library/react';
import App from './App';


describe('Markdown Preview', () => {
  it('Should render correctly', () => {
    const tree = renderer
      .create(<App />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should depict changes from input', () => {
    const { baseElement } = render(<App />);

    const input = screen.getByText(/# markdown preview/i);
    fireEvent.change(input, { target: { value: 'test markDown' } });
    
    expect(baseElement);
  });
});
