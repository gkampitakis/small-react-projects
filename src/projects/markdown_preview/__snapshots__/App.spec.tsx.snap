// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown Preview Should render correctly 1`] = `
<main
  className="markdown_preview"
>
  <section>
    <textarea
      onChange={[Function]}
      value="# markdown preview"
    />
    <article>
      <div
        className="markdown-body"
      >
        <h1>
          markdown preview
        </h1>
      </div>
    </article>
  </section>
</main>
`;
